app/utils/risk_index.py

```diff
@@ -0,0 +1,81 @@
+from typing import List, Optional
+
+from sqlalchemy.orm import Session
+import pandas as pd
+
+import crud
+import utils
+import uuid
+
+from models import PestModel
+
+
+def calculate_risk_index_probability(db: Session, weather_dataset_id: int, pest_models: List[PestModel], parameter: Optional[str] = None):
+    # SQL query for the data
+    data_db = crud.data.get_data_interval_query_by_dataset_id(db=db, dataset_id=weather_dataset_id)
+
+    df = pd.read_sql(sql=data_db.statement, con=db.bind, parse_dates={"date": "%Y-%m-%d"})
+
+    # Calculate the risks associated with each pest_model
+    for pm in pest_models:
```

Comment:
Here we have many nested `for` but I suppose won't be a problem taking into account the no of rules per pest_model